HyperVExtension/src/DevSetupAgent/DevAgentService.cs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using HyperVExtension.Telemetry;
+using Microsoft.Windows.AppLifecycle;
+
+namespace HyperVExtension.DevSetupAgent;
+
+/// <summary>
+/// Service to communicate between host and guest machines.
+/// The main loop waits for messages from the host, processes them, and send response back to host.
+/// </summary>
+public class DevAgentService : BackgroundService
+{
+    private readonly IHost _host;
+
+    public DevAgentService(IHost host)

```

Comment:
Are `IHostChannel` and `IRequestManager` singleton? If so avoid directly injecting `IHost`, instead directly inject `IHostChannel` and `IRequestManager`. If they are transient, consider registering a factory class/function.